data_prep/scotland/src/bin/generate_prioritization.rs

```diff
@@ -235,61 +237,60 @@ impl SettlementsInput {
 }
 
 fn calculate_metric_buckets(
-    data: &mut ContextData,
-    population_zone_area_km2: Vec<f64>,
-) -> Result<()> {
-    data.metric_buckets.population_density = make_buckets(
-        &data
-            .population_zones
+    population_zone_inputs: &[PopulationZoneInput],
+    stats19_collisions: &[Point],
+    pois: &[POI],
+) -> Result<MetricBuckets> {
+    let population_density = make_buckets(
+        &population_zone_inputs
             .iter()
-            .enumerate()
-            .map(|(idx, zone)| zone.population as f64 / population_zone_area_km2[idx])
+            .map(|zone| zone.population as f64 / zone.area_km2())
             .collect(),
     )?;
 
-    let population_zones = data
-        .population_zones
-        .iter()
-        .map(|zone| PreparedGeometry::from(&zone.geometry))
-        .collect::<Vec<_>>();
-
-    let mut collions_per_zone = Vec::new();
-    for zone in &population_zones {
+    let mut collisions_per_zone = Vec::new();
+    for zone in population_zone_inputs {
```

Comment:
Just a note to self: before, the population zones passed in were pre-filtered to intersect the study area boundary. Now we pass all of them in. But the point-in-polygon test still behaves correctly, so works fine